bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/texteditor/FindReplaceOverlay.java

```diff
@@ -0,0 +1,1000 @@
+				toggleToolItem(searchInSelectionButton);
+			} else if ((e.stateMask & SWT.CTRL) != 0 && (e.keyCode == 'C' || e.keyCode == 'c')) {
+				toggleToolItem(caseSensitiveSearchButton);
+			} else if (e.keyCode == SWT.CR || e.keyCode == SWT.KEYPAD_CR) {
+				performEnterAction(e);
+			} else if (e.keyCode == SWT.ESC) {
+				close();
+			} else {
+				e.doit = true;
+			}
+		}
+
+		private void toggleToolItem(ToolItem toolItem) {
+			toolItem.setSelection(!toolItem.getSelection());
+			toolItem.notifyListeners(SWT.Selection, null);
+		}
+
+		@Override
+		public void keyReleased(KeyEvent e) {
+			// Do nothing
+		}
+	};
+
+	private ControlListener shellMovementListener = new ControlListener() {
+		@Override
+		public void controlMoved(ControlEvent e) {
+			positionToPart();
+		}
+
+		@Override
+		public void controlResized(ControlEvent e) {
+			positionToPart();
+		}
+	};
+	private FocusListener overlayFocusListener = new FocusListener() {
+
+		@Override
+		public void focusGained(FocusEvent e) {
+			// do nothing
+		}
+
+		@Override
+		public void focusLost(FocusEvent e) {
+			findReplaceLogic.activate(SearchOptions.GLOBAL);
+			searchInSelectionButton.setSelection(false);
+		}
+
+	};
+	private PaintListener widgetMovementListener = new PaintListener() {
+
+		@Override
+		public void paintControl(PaintEvent e) {
+			positionToPart();
+		}
+
+	};
+	private IPartListener partListener = new IPartListener() {
+		@Override
+		public void partActivated(IWorkbenchPart part) {
+			if (getShell() != null) {
+				getShell().setVisible(isPartCurrentlyDisplayedInPartSash());
+			}
+		}
+
+		@Override
+		public void partDeactivated(IWorkbenchPart part) {
+			// Do nothing
+		}
+
+		@Override
+		public void partBroughtToTop(IWorkbenchPart part) {
+			if (getShell() != null) {
+				getShell().setVisible(isPartCurrentlyDisplayedInPartSash());
+			}
+		}
+
+		@Override
+		public void partClosed(IWorkbenchPart part) {
+			close();
+		}
+
+		@Override
+		public void partOpened(IWorkbenchPart part) {
+			// Do nothing
+		}
+	};
+
+	private boolean isPartCurrentlyDisplayedInPartSash() {
+		IWorkbenchPage activePage = PlatformUI.getWorkbench().getActiveWorkbenchWindow().getActivePage();
+
+		// Check if the targetPart is currently displayed on the active page
+		boolean isPartDisplayed = false;
+
+		if (activePage != null) {
+			IWorkbenchPart activePart = activePage.getActivePart();
+			if (activePart != null && activePart == targetPart) {
+				isPartDisplayed = true;
+			}
+		}
+
+		return isPartDisplayed;
+	}
+
+	/**
+	 * Returns the dialog settings object used to share state between several
+	 * find/replace overlays.
+	 *
+	 * @return the dialog settings to be used
+	 */
+	private static IDialogSettings getDialogSettings() {
+		IDialogSettings settings = PlatformUI
+				.getDialogSettingsProvider(FrameworkUtil.getBundle(FindReplaceOverlay.class)).getDialogSettings();
+		return settings;
+	}
+
+	@Override
+	public void create() {
+		super.create();
+	}
```

Comment:
Can't this method be removed?

Reply:
Yes, it can. We want the same things to happen when the overlay is created first as when the overlay is re-opened from the Editor.

Reply:
removed in https://github.com/eclipse-platform/eclipse.platform.ui/commit/2f9136a25069244769073ee77ad5be15b5dadeee